version: "3.9"
services:
  api:
    image: $DOCKER_USERNAME/$API_NAME:$GITHUB_REF_NAME-$GITHUB_RUN_ID
    ports:
      - 25000:25000
    volumes:
      - ./docker_config.py:/app_code/config.py
    depends_on:
      pg_main_db:
        condition: service_healthy
      migrations:
        condition: service_completed_successfully
  front:
    image: $DOCKER_USERNAME/$FRONT_NAME:latest
    ports:
      - 80:2500
    depends_on:
      pg_main_db:
        condition: service_healthy
      migrations:
        condition: service_completed_successfully

  migrations:
    image: $DOCKER_USERNAME/$API_NAME:$GITHUB_REF_NAME-$GITHUB_RUN_ID
    environment:
      FLASK_APP: 'run.py'
    volumes:
      - ./docker_config.py:/app_code/config.py
    depends_on:
      pg_main_db:
        condition: service_healthy
    command: sh -c "python3 -m flask db upgrade"

  pg_main_db:
    image: postgres:latest
    environment:
      POSTGRES_USER: $MAIN_DB_USER
      POSTGRES_PASSWORD: $MAIN_DB_PASS
      POSTGRES_DB: $MAIN_DB_NAME
    volumes:
      - ./pg_main_data:/var/lib/postgresql/data
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U $MAIN_DB_USER -d $MAIN_DB_NAME" ]
      interval: 5s
      timeout: 5s
      retries: 5
